File: backend/payment-service/src/main/java/ecommerce/paymentservice/entity/Payment.java
package ecommerce.paymentservice.entity;

import ecommerce.apicommon1.model.status.PaymentMethodStatus;
import ecommerce.apicommon1.model.status.PaymentStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "PAYMENT", schema = "order_db")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "payment_seq")
    @SequenceGenerator(name = "payment_seq", sequenceName = "SEQ_PAYMENT_ID", allocationSize = 1)
    @Column(name = "PAYMENT_ID")
    private Long id;

    @Column(name = "USER_ID")
    private Long userId;

    @Column(name = "ORDER_ID")
    private Long orderId;

    @Column(name = "ORDER_CODE")
    private String orderCode;

    @Column(name = "PAYMENT_DATE")
    private Instant paymentDate;

    @Column(name = "AMOUNT_PAID")
    private BigDecimal amountPaid;


    @Enumerated(EnumType.STRING)
    @Column(name = "PAYMENT_METHOD")
    private PaymentMethodStatus paymentMethod;

    @Enumerated(EnumType.STRING)
    @Column(name = "STATUS")
    private PaymentStatus status;

    @CreationTimestamp
    @Column(name = "CREATED_AT")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "UPDATED_AT")
    private LocalDateTime updatedAt;

    @Transient
    private String checkoutUrl;

    @Transient
    private String stripePaymentIntentId;
}

File: backend/payment/src/entity/Payment.js
const {EntitySchema} = require('typeorm');

module.exports = new EntitySchema({
    name: 'Payment',
    tableName: 'PAYMENT',
    columns: {
        skuCode: {
            type: Number,
            primary: true,
            name: 'PAYMENT_ID',
        },
        orderId: {
            type: Number,
            name: 'ORDER_ID',
        },
        orderCode: {
            type: String,
            name: 'ORDER_CODE',
        },
        userId: {
            type: Number,
            name: 'USER_ID',
        },
        paymentDate: {
            type: 'timestamp',
            name: "PAYMENT_DATE",
        },
        amountPaid: {
            type: Number,
            name: "AMOUNT_PAID",
        },
        paymentMethod: {
            type: String,
            name: "PAYMENT_METHOD",
        },
        status: {
            type: String,
            name: "STATUS",
        },
        createdAt: {
            type: 'timestamp',
            name: "CREATED_AT",
            createDate: true,
            default: () => 'CURRENT_TIMESTAMP',
        },
        updatedAt: {
            type: 'timestamp',
            name: "UPDATED_AT",
            updatedAt: true,
            onUpdate: () => 'CURRENT_TIMESTAMP',
        }
    }
});
